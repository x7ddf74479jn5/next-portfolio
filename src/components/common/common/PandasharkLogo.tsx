import Image from "next/image";
import Link from "next/link";

import { images } from "../../../utils/images";

type Props = {
  width: number;
  height: number;
};

const PandasharkLogo: React.VFC<Props> = ({ width, height }) => {
  return (
    <>
      <Link href="/">
        <a target="_self" aria-label="パンダシャーク">
          <Image
            width={width}
            height={height}
            alt={images.pandasharkRectangle.alt}
            src={images.pandasharkRectangle.path}
          />
        </a>
      </Link>
    </>
  );
};

export default PandasharkLogo;
