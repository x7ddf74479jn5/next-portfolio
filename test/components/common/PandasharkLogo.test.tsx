import { render } from "@testing-library/react";
import PandasharkLogo from "src/components/common/PandasharkLogo";

jest.mock("next/image", () => {
  const Image = ({
    src,
    alt,
    width,
    height,
  }: {
    src: string;
    alt: string;
    width: number;
    height: number;
  }): JSX.Element => {
    return <img src={src} alt={alt} width={width} height={height} />;
  };
  return Image;
});

describe("PandasharkLogo", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<PandasharkLogo width={128} height={64} />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it("Next Link component compiles is transformed the anchor element with appropriate href", () => {
    const { container } = render(<PandasharkLogo width={128} height={64} />, {});
    expect(container.firstChild).toHaveAttribute("href", "/");
  });

  it("has alt", () => {
    const { container, debug } = render(<PandasharkLogo width={128} height={64} />, {});
    const targetNode = container.firstChild?.firstChild;
    expect(targetNode).toHaveAttribute("alt", expect.stringContaining("Pandashark"));
    debug();
  });
});
