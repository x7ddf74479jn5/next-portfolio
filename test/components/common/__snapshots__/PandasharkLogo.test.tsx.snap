// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PandasharkLogo matches snapshot 1`] = `
<DocumentFragment>
  <a
    aria-label="パンダシャーク"
    href="/"
    target="_self"
  >
    <img
      alt="Pandashark's icon"
      height="64"
      src="/img/icons/pandashark_logo_rectangle.webp"
      width="128"
    />
  </a>
</DocumentFragment>
`;
